src/test/config.test.ts

```diff
@@ -339,14 +339,25 @@ test('should load cors origin, maxAge and exposedHeaders default values', () =>
     expect(config.cors.exposedHeaders).toBe('ETag');
 });
 
-test('should load passed in http agent', () => {
+test('when passed with agent in httpOptions config.httpOptions.agent should be callable with url', () => {
     const config = createProxyConfig({
         unleashUrl: 'some',
         unleashApiToken: 'some',
         clientKeys: ['s1'],
         httpOptions: {
-            agent: _ => https.globalAgent
-        }
+            agent: (_) => https.globalAgent,
+        },
     });
-    expect(config.httpOptions?.agent?.(new URL('https://example.com'))).toBe(https.globalAgent);
-})
\ No newline at end of file
+    expect(config.httpOptions?.agent?.(new URL('https://example.com'))).toBe(
+        https.globalAgent,
+    );
+});
+
+test('when not passed with httpOptions the config should not contain that property', () => {
```

Comment:
Minor test description rewording suggestion 😄 

```suggestion
test('should not set config.httpOptions if no http options are provided at creation', () => {
```